src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -728,31 +728,21 @@ public void SetLocation(Geometry geometry)
         {
             TransactionManager.Instance.EnsureInTransaction(Application.Document.Current.InternalDocument);
 
-            if (this.InternalElement.Location.GetType() == typeof(Autodesk.Revit.DB.LocationPoint))
+            if (this.InternalElement.Location.GetType() == typeof(Autodesk.Revit.DB.LocationPoint) && typeof(Autodesk.DesignScript.Geometry.Point).IsAssignableFrom(geometry.GetType()))
```

Comment:
You could have done the casting before the if() condition and then check for null. Here type check is happening twice.
```
var locationPoint = this.InternalElement.Location as Autodesk.Revit.DB.LocationPoint;
var point = geometry as Autodesk.DesignScript.Geometry.Point;
if(locationPoint != null && point != null)
{
    locationPoint.Point = point.ToRevitType(true);
}
```